src/asm/macro.c

```diff
@@ -98,6 +98,35 @@ char const *macro_GetArg(uint32_t i)
 					      : macroArgs->args[realIndex];
 }
 
+char const *macro_GetAllArgs(void)
+{
+	if (!macroArgs)
+		return NULL;
+
+	/*
+	 * Returning an empty string for no arguments
+	 * does not cooperate with beginExpansion,
+	 * so this returns a space.
+	 */
+	if (macroArgs->shift >= macroArgs->nbArgs)
+		return strdup(" ");
```

Comment:
Wouldn't it be better to return `NULL`, and have the caller not perform any expansion if that's returned?

Reply:
`peek` calls `expandMacroArg` to "Expand the argument and return its first character". If no expansion is performed, `expandMacroArg` would have to return a sentinel `NULL`, and `peek` would have to continue on to the next character. Does that sound better? It has more special handling for the consequences of `\#` expanding to nothing, but avoids the arbitrary expansion to `" "`.

Reply:
I think it'd be less overhead than allocating an expansion and then all the plumbing related to reading over it, so please go ahead with the former.

Reply:
Okay, done. This would also handle any other macro args which somehow expand to empty strings.